### DISCUSSION TITLE
Centralise the JavaVersion

### DISCUSSION THREAD

**Discussion by mcatta** [androiddevelopersitalia/aditogether#17] [General]
I noticed that in a couple of files we are using:
https://github.com/androiddevelopersitalia/aditogether/blob/fe6c637ce9904fb4100a1c193d6f4eebd935b969/build-tools/lint/src/main/kotlin/aditogether/buildtools/lint/LintPlugin.kt#L23-L26
and here:
https://github.com/androiddevelopersitalia/aditogether/blob/fe6c637ce9904fb4100a1c193d6f4eebd935b969/playground/build.gradle#L6-L7
we are using the same `JavaVersione`so I'm wondering if could make sense to centralize that information in a single place.

What do you think @fondesa @4face-studi0? Considering also the growing of modules. 

In case of yes, given that we don't use the `buildSrc/dept.kt` file what could be a good place to set that constant? build-tools?

**Comment by fondesa** [comment]
Those in playground are just temporary.
The source and target compatibility are related to the compilation of JVM and Android sources. So they should belong to the same shared plugin.
Detekt needs to infer those from this shared plugin as well IMHO.

Currently we don't have these plugins because they should be created based on our future needs, but yes, these constants will be 100% centralized in only one place.

**Comment by fardavide** [comment]
As @fondesa said.

In a near future, I would expect to see only these things in a `build.gradle` file:
1) application of our plugins
2) dependencies 
3) minimal config of our plugins if needed 

And nothing more than that

**Comment by mcatta** [comment]
Sounds great, thank you guys!

**Comment by fondesa** [comment]
This was centralized, so I'm closing this
